pvlib/irradiance.py

```diff
@@ -169,6 +170,11 @@ def aoi_projection(surface_tilt, surface_azimuth, solar_zenith, solar_azimuth):
         Solar zenith angle.
     solar_azimuth : numeric
         Solar azimuth angle.
+    projection_minimum : None or numeric, default None
+        Minimum allowable value for the projection for the solar angle
+        into the plane of the array. For example, 0 is equivalent to a
+        90 degree limit, 0.01745 is equivalent to a 89 degree limit.
+        `None` applies no limit.
 
```

Comment:
Yes, I also wondered this. I only went with the projection because it's the quantity that is hardcoded in the matlab library.

Reply:
After thinking through the response to your comment below, I am leaning towards keeping it as the dot product to avoid the complexity of handling angles > 90 degrees or less than 0.

Reply:
I'm having trouble seeing a use case which requires `projection_minimum`.  The function's description says it calculates a dot product so that's what I'd expect it to do, and to not do more.

Reply:
Actually it does *less* than a dot product since it assumes/requires the vectors are *unit* vectors.  It's really just calculating the cosine of the angle of incidence.  I think the name "projection" seems to suggest more, and may even be confusing to some.

Reply:
I will remove the ``projection_minimum`` kwargs and put the limits in the diffuse models where necessary. We can address the function name in a separate issue.